Mod2/Lessons/Week2/DataRelationships.md

```diff
@@ -0,0 +1,158 @@
+
+But, what if we want to keep track of the artist that wrote each song?  We _could_ add a column to our database called `artist` and store the artist directly on a song, like this:
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_2.png'>
+</p>
+
+Seems like it works, right? Well... not exactly.  What if we wanted to also store information about specific artists, like their hometown or years active.  If we imagine what that might look like, we would see more columns on our `songs` table and many of those columns would be storing the exact same information - every Prince song would be storing 'Minneapolis' and '1975-2016':
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_3.png'>
+</p>
+
+This is very repetitive!
+
+## Normalizing our Database
+
+In order to cut down on redundant data on a specific table in our database, we can rely on **normalization**.  Normalization is a process by which we break down tables in our database so that each table relates only to one resource.  In the example we are working with, when we look at our songs table, we are storing information about songs and about artists. Really, we are trying to store information about two resources on one table - let's fix that.
+
+In your pairs, draw how you would normalize this database, so that songs and artists each have their own table. I highly recommend you look back at the Intro to Databases lesson to see how we normalized customers and orders!
+
+When we get back, be prepared to share your diagrams and answer the following questions.
+
+* How are you able to still know which artist wrote which song?
+* What primary key do you have?
+* What foreign key do you have?
+
+<details><summary>Normalized Songs and Artists Database</summary><br/>
+
+When we normalize our database, we need to include some sort of marker for ourselves to know how to relate the tables to one another.  We do this with **primary keys** and **foreign keys**.  As our database exists now, every record has an `id`.  We can use these ids to relate information in one table, to information in the other (or give each of our songs an artist).  For each song in our database, we will add an attribute `artist_id` that will hold the `id` of the artist that song belongs to.
+
+<p align='center'>
+  <img src='../../Images/Week2/1_to_many_db_image_5.png'>
+</p>
+
+This `artist_id` is the **foreign key** on our songs table that relates a song back to the `id` of an artist on the artists table - also referred to as the **primary key** of the artists table.
+
+</details>
+</br>
+As we build out our databases, we will want them to be as normalized as possible - only storing information from a single resource per table.
+
+## Describing Data Relationships
+
+Learning the language to describe database relationships can feel tedious, but having consistency with how we describe these relationships will help us communicate our database design to other developers.
+
+### One-to-Many
```

Comment:
Do you prefer to describe `One-to-Many` before `One-to-One`? I feel 1:1 is easier to describe and understand. Students can probably think of more examples of 1:1, and mistakenly describe a 1:M relationship as 1:1, which would be a nice segue into 1:M. Just thinking out loud and typing as I think 😄

Reply:
I agree, 1:1 does feel easier to understand. I would normally start with that, but my thought process behind one-to-many first here is because that's the relationship we had been talking about earlier in the lesson. So was thinking we'd start there, and then continue on to the others.